src/Microsoft.VisualStudio.Extensibility.Testing.Xunit.Shared/Threading/IdeTestCase.cs

```diff
@@ -83,8 +83,8 @@ public override void Serialize(IXunitSerializationInfo data)
 
         public override void Deserialize(IXunitSerializationInfo data)
         {
-            base.Deserialize(data);
             VisualStudioVersion = (VisualStudioVersion)data.GetValue<int>(nameof(VisualStudioVersion));
+            base.Deserialize(data);
```

Comment:
Why swap these lines?

Reply:
This is part of this fix:

> Fix failure to report the correct display name for skipped tests (showed Unspecified instead of a specific version)

The call to `base.Deserialize` ends up calling `GetDisplayName`, so we need to initialize `VisualStudioVersion` before that. We can't move `SkipReason` because the setter triggers the call to `GetDisplayName` expecting the call to `base.Deserialize` to already be complete.

Reply:
Makes sense